cirq/experiments/cross_entropy_benchmarking.py

```diff
@@ -0,0 +1,363 @@
+import numpy as np
+from typing import List, Set, Tuple, Sequence, Dict, Any, NamedTuple, Union, \
+    Iterable
+
+from matplotlib import pyplot as plt
+from cirq import devices, ops, circuits, sim
+
+CrossEntropyPair = NamedTuple('CrossEntropyPair', [('num_cycle', int),
+                                                   ('xeb_fidelity', float)])
+
+
+class CrossEntropyResult:
+    """Results from a cross-entropy benchmarking (XEB) experiment."""
+
+    def __init__(self, cross_entropy_pair: Sequence[CrossEntropyPair]):
+        """
+        Args:
+            cross_entropy_pair: A sequence of NamedTuples, each of which
+                contains two fields: num_cycle which returns the circuit
+                depth as the number of cycles and xeb_fidelity which returns
+                the XEB fidelity after the give cycle number.
+        """
+        self._data = cross_entropy_pair
+
+    @property
+    def data(self) -> Sequence[CrossEntropyPair]:
+        """Returns a sequence of NamedTuples, each of which contains a cycle
```

Comment:
Summary should be one line, then perhaps more details in a paragraph.

Reply:
Done.